lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3OutputStream.java

```diff
@@ -195,6 +212,7 @@ private void flushBuffer(boolean finished)
         if (finished && !multipartUploadStarted) {
             PutObjectRequest request = PutObjectRequest.builder()
                     .overrideConfiguration(context::applyCredentialProviderOverride)
+                    .acl(cannedAcl)
```

Comment:
we should set the `acl` only if not `null`

Reply:
I dont think we follow that format for other configs. eg see sseType

Reply:
I think it's fine to always set it, as the API is designed to accept null as a way to not set an ACL on the upload.